stats/opentelemetry/server_metrics.go

```diff
@@ -0,0 +1,191 @@
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.call.started\", will not record") // error or log?
+		} else {
+			registeredMetrics.serverCallStarted = scs
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.server.call.sent_total_compressed_message_size"]; ok {
+		ss, err := meter.Int64Histogram("grpc.server.call.sent_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) sent across all response messages (metadata excluded) per RPC; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.call.sent_total_compressed_message_size\", will not record") // error or log?
+		} else {
+			registeredMetrics.serverCallSentTotalCompressedMessageSize = ss
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.server.call.rcvd_total_compressed_message_size"]; ok {
+		sr, err := meter.Int64Histogram("grpc.server.call.rcvd_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) received across all request messages (metadata excluded) per RPC; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.rcvd.sent_total_compressed_message_size\", will not record")
+		} else {
+			registeredMetrics.serverCallRcvdTotalCompressedMessageSize = sr
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.server.call.duration"]; ok {
+		scd, err := meter.Float64Histogram("grpc.server.call.duration", metric.WithUnit("s"), metric.WithDescription("This metric aims to measure the end2end time an RPC takes from the server transport’s perspective."), metric.WithExplicitBucketBoundaries(DefaultLatencyBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.server.call.duration\", will not record")
+		} else {
+			registeredMetrics.serverCallDuration = scd
+		}
+	}
+
+	ssh.registeredMetrics = registeredMetrics
+}
+
+// TagConn exists to satisfy stats.Handler.
+func (ssh *serverStatsHandler) TagConn(ctx context.Context, _ *stats.ConnTagInfo) context.Context {
+	return ctx
+}
+
+// HandleConn exists to satisfy stats.Handler.
+func (ssh *serverStatsHandler) HandleConn(context.Context, stats.ConnStats) {}
+
+// TagRPC implements per RPC context management.
+func (ssh *serverStatsHandler) TagRPC(ctx context.Context, info *stats.RPCTagInfo) context.Context {
+	method := info.FullMethodName
+	if ssh.mo.MethodAttributeFilter != nil {
+		if !ssh.mo.MethodAttributeFilter(method) {
+			method = "other"
+		}
+	}
+	server := internal.ServerFromContext.(func(context.Context) *grpc.Server)(ctx)
+	if server == nil { // Shouldn't happen, defensive programming.
```

Comment:
As above: probably log error && remove comment.

Reply:
It should be clear from the log message that it is unexpected.  So the comments shouldn't be necessary.

Reply:
In general we should minimize our comments by making the code better.  Logging/returning an error can be all the commentary you need in most error cases.

Reply:
Removed.